src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/AttachedCollectionItemBase.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Windows;
+using Microsoft.Internal.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Imaging.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections
+{
+    /// <summary>
+    /// Base class for attached items in the dependencies tree in Solution Explorer.
+    /// </summary>
+    internal abstract class AttachedCollectionItemBase
+        : ITreeDisplayItem,
+          ITreeDisplayItemWithImages,
+          IPrioritizedComparable,
+          IInteractionPatternProvider,
+          IBrowsablePattern
+    {
+        // Other patterns we may wish to utilise in future are:
+        //
+        // - IInvocationPattern
+        // - IContextMenuPattern
+        // - ISupportExpansionEvents
+        // - ISupportExpansionState
+        // - IDragDropSourcePattern
+        // - IDragDropTargetPattern
+        // - ISupportDisposalNotification
+        // - IRenamePattern
+        // - We also see requests for IVsHierarchyItem
+        //
+        // NOTE we don't have to support ITreeDisplayItemWithImages -- it's covered by ITreeDisplayItem
+
+        private static readonly HashSet<Type> s_supportedPatterns = new HashSet<Type>
+        {
+            typeof(ITreeDisplayItem),
+            typeof(IBrowsablePattern),
+        };
+
+        protected AttachedCollectionItemBase(string name) => Text = name;
+
+        public string Text { get; }
+
+        public abstract int Priority { get; }
+
+        public virtual FontStyle FontStyle => FontStyles.Normal;
+
+        public virtual FontWeight FontWeight => FontWeights.Normal;
+
+        public abstract ImageMoniker IconMoniker { get; }
+
+        public virtual ImageMoniker ExpandedIconMoniker => IconMoniker;
+
+        public virtual ImageMoniker OverlayIconMoniker => default;
+
+        public virtual ImageMoniker StateIconMoniker => default;
+
+        public virtual string? StateToolTipText => null;
+
+        // Return null means ToolTipText is displayed only when the item's label is truncated
+        public object? ToolTipContent => null;
+
+        public string ToolTipText => Text;
+
+        public bool IsCut => false;
+
+        public virtual object? GetBrowseObject() => null;
+
+        public TPattern? GetPattern<TPattern>() where TPattern : class
+        {
+            if (s_supportedPatterns.Contains(typeof(TPattern)))
```

Comment:
Is this even worth doing? Just `return this as TPattern` will have the same effect, right? Or do you specifically want to return false for `ITreeDisplayItemWithImages`? If so, perhaps the comment above should be more strongly worded than "we don't have to support"

Reply:
Possibly not needed. This PR is a stripped down version of work I have locally in which this method behaves differently when the item is disposed (in which case you want to support a subset of patterns). There's a chance that won't be needed in the end.

Most (all?) [other implementations of this member](http://index/?leftProject=Microsoft.VisualStudio.Shell.Framework&leftSymbol=a2bwcgbamcgc&rightProject=Microsoft.Ide.ProjectSystem&file=Hierarchy%5CHierarchyItem.cs&line=328) call the patterns out explicitly.

Reply:
You pre-empted the only comment I have on this PR 😛